integrate/simpsons_test.go

```diff
@@ -0,0 +1,82 @@
+// Copyright ©2019 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package integrate
+
+import (
+	"math"
+	"testing"
+
+	"gonum.org/v1/gonum/floats"
+)
+
+func TestSimpsons(t *testing.T) {
+	const N = 1e6
+	x := floats.Span(make([]float64, N), 0, 1)
```

Comment:
This should create unevenly spaced locations, it would have revealed the mistakes in the formulas which do not manifest themselves with uniform subdivison.

Reply:
Thanks.

The issue I'm struggling with is supplying a passable test that jointly illustrates these points and also serves as a 'true' representation of the algorithm. It's my understanding that all common methods used to perform numerical integrations make the assumption that points are evenly distributed.

My take is that users should be responsible for passing values that are suitable for the mathematical model and form (e.g Newton-Cotes) over implementing additional computational arithmetic.

That being said, I'm not sure if segmentation over irregularly spaced data (http://cs-people.bu.edu/evimaria/cs565/TT06.pdf) where one could segment intervals and then evaluate the integral, is a possible solution? Evaluating x in any condition is costly in and of itself. 

```
// Apply Simpson's to the k-segment produced from ...
{
			x:    append(floats.Span(make([]float64, N), 0, 1),
                                            floats.Span(make([]float64, N), 100, 101)...),
			f:    func(x float64) float64 { return x },
			want: 50.5,
		},
```

Reply:
This code allows non-uniform spacing, so the test should generate such input. You can either randomly jitter values from `floats.Span` or generate `x` algorithmically.